### ISSUE TITLE
Implement `CrossEntropyLoss`

### ISSUE THREAD

**Issue by joewallwork** [Cambridge-ICCS/FTorch#588]
I need this for the work on predicting number of halving steps.

**Comment by joewallwork** [comment]
See https://docs.pytorch.org/cppdocs/api/nn/loss.html#_CPPv4N5torch2nn16CrossEntropyLossE.

**Comment by joewallwork** [comment]
See also https://docs.pytorch.org/docs/main/nn.functional.html#torch.nn.functional.mse_loss and https://docs.pytorch.org/cppdocs/search.html?q=mseloss#.
